### PR TITLE
Azure/vpnconnection cleaner

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/3882

Provide Azure vpn connection cleanup

### CODE REVIEW

File: pkg/cleaner/azure/vpnconnection.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+					continue
+				}
+
+				c.logger.Log("level", "error", "message", fmt.Sprintf("ensured deletion of vpn connection %q", *connection.Name))
+			}
+		}
+	}
+
+	if lastError != nil {
+		return microerror.Mask(lastError)
```
Comment (MarcelMue) [inline pkg/cleaner/azure/vpnconnection.go]: I don't really follow the reason for this. Why do we return only the last error? And not right away when we can't delete one?

File: cmd/azure.go [hunk 2]
```diff
@@ -116,3 +117,9 @@ func newVirtualNetworksClient(azureSubscriptionID string, servicePrincipalToken
 
 	return &c
 }
+func newVirtualNetworkGatewayConnectionsClient(azureSubscriptionID string, servicePrincipalToken *adal.ServicePrincipalToken) *network.VirtualNetworkGatewayConnectionsClient {
```
Comment (MarcelMue) [inline cmd/azure.go]: I still find the naming in azure packages weird. It feels like they have a competition for the longest possible name for something. (nothing you can do about this though)

**Review by MarcelMue — COMMENTED** [review]
I am not too familiar with Azure, this seems pretty solid though. A second opinion would be good.

File: cmd/azure.go (hunk 2 above)
Comment (TheoBrigitte) [reply cmd/azure.go]: Would `VNetGatewayConnections` help ? 10 characters shorter, not bad

File: pkg/cleaner/azure/vpnconnection.go (hunk 1 above)
Comment (TheoBrigitte) [reply pkg/cleaner/azure/vpnconnection.go]: The logic here is to clean up as much resources as possible, so whenever we encounter an error we store it and only return it at the end. A more accurate solution would be to collect all errors and return a single error containing all collected errors.

**Review by fgimenez — APPROVED** [review]
LGTM, I like the separate files for each resource.

File: cmd/azure.go (hunk 2 above)
Comment (MarcelMue) [reply cmd/azure.go]: Don't worry about it, this is fine.

File: pkg/cleaner/azure/vpnconnection.go (hunk 1 above)
Comment (MarcelMue) [reply pkg/cleaner/azure/vpnconnection.go]: I see, Pawel did the collection thing - I am not sure if it's beneficial here:
https://github.com/giantswarm/aws-operator/blob/1fc2ae9aa574018eee5e0c1da4e8da2a295d0142/integration/setup/teardown.go#L17-L57

**Review by MarcelMue — APPROVED** [review]
Fine by me

File: pkg/cleaner/azure/vpnconnection.go (hunk 1 above)
Comment (TheoBrigitte) [reply pkg/cleaner/azure/vpnconnection.go]: In the end he only returns the first error. First or last does not make any difference.
We can improve on this later on when we feel the need.

**Pull request merged**